### PR TITLE
guide: updates to `dvc.yaml` spec

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Extracted from https://github.com/iterative/dvc.org/pull/3414#issuecomment-1189898267

This page should focus on the schema/spec of the file and it's sections more than guidance and explaining concepts, since we'll have a guide for that now (see https://github.com/iterative/dvc.org/pull/3414).

### CODE REVIEW

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 1]
```diff
@@ -1,29 +1,16 @@
...
 
-You can construct data science or machine learning pipelines by defining
-individual [stages](/doc/command-reference/run) in one or more `dvc.yaml` files.
-Stages form a pipeline when they connect with each other (forming a _dependency
-graph_, see `dvc dag`). Refer to
-[Get Started: Data Pipelines](/doc/start/data-pipelines).
+You can construct machine learning pipelines by defining individual
+[stages](/doc/command-reference/run) in one or more `dvc.yaml` files. Stages
+constitute a pipeline when they connect with each other (forming a [dependency
+graph], see `dvc dag`).
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:6]: Only the changes to this file should exist in this PR... Weird 🤔

**Comment by shcheklein** [comment]
@jorgeorpinel what is the status for this?

**Comment by jorgeorpinel** [comment]
Ready for review @shcheklein but should probably not be merged before #3414, whence it came.

> #3414 is also ready for review but it's too big again so I need to split it more and didn't have time yet, sorry.

**Comment by dberenbaum** [comment]
@jorgeorpinel What's the status of this PR? Is it still needed?

**Comment by jorgeorpinel** [comment]
I'm not sure how this PR got so complicated 😓 but yes, after merging master (resolved a bunch of conflicts) I see many useful changes here. I will have to split this though, otherwise it's too hard to review... ⏳

File: content/docs/user-guide/project-structure/dvcyaml-files.md [hunk 2]
```diff
@@ -206,6 +202,65 @@ stages:
 The commands in `dvc metrics` and `dvc plots` help you display and compare
 metrics and plots.
 
+## Stage entries
+
+These are the fields that are accepted in each stage:
+
+| Field            | Description                                                                                                                                                                                                                                                                                     |
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvcyaml-files.md:209]: All this I just moved up unchanged. It started in old line 506 before.

**Review by dberenbaum — APPROVED** [review]
Thanks @jorgeorpinel! LGTM except confused why it says 88 commits?

It would help me better manage the docs workload if we can be a little more careful about what changes are worthwhile. If the PR was all about making it more of a spec and less of a guide it would make more sense, but most changes look like minor, subjective style issues. If we can reduce some of those, I hope it can help reduce noise and focus more on what's important. I know you don't intend for me to have to review everything right away, but it would help to cut out the unimportant stuff altogether instead of having to try to figure it out as I review the PR. 🙏 

What do you think?

**Comment by jorgeorpinel** [comment]
It's 88 commits because this was started in July. I later removed a bunch of changes.

Agree about the relevance of these changes. At that time I already had a bunch of edits stashed for this file which I didn't manage to propose since they never seemed major enough so eventually I just made the PR (and then it grew out of control).

But don't worry, in the new docs process, this kind of work is not something I'll be paying much attention to in the foreseeable future.

**Pull request merged**